### PR TITLE
Feature/trl 12 declaration

### PR DESCRIPTION
**Pull request by ShahanAliyev**

### CODE REVIEW

File: order/admin.py [hunk 1]
```diff
@@ -1,6 +1,12 @@
...
 
 admin.site.register(Status)
 
-# Register your models here.
+
+@admin.register(Declaration)
+class DeclarationAdmin(admin.ModelAdmin):
+
+    def save_model(self, request, obj, form, change):
+        obj.user = request.user
```
Comment (mmzeynalli) [inline order/admin.py]: Let admin be able to choose user as well. If no user is defined, then user should be THAT user

File: order/apps.py [hunk 2]
```diff
@@ -4,3 +4,6 @@
 class OrderConfig(AppConfig):
     default_auto_field = 'django.db.models.BigAutoField'
     name = 'order'
+
+    def ready(self):
+        import order.signals
```
Comment (mmzeynalli) [inline order/apps.py:9]: why?

File: order/models.py [hunk 3]
```diff
@@ -8,4 +15,65 @@ class Status(models.Model):
...
+    discount = models.ManyToManyField(Discount, related_name='declarations', blank=True)
+    weight = models.DecimalField(max_digits=5, decimal_places=3, null=True, blank=True)
+    is_paid = models.BooleanField(default=False)
+    penalty_status = models.BooleanField(default=False)
+    penalty = models.DecimalField(max_digits=4, decimal_places=2, default=0, null=True, blank=True)
+
+    
+    def save(self, *args, **kwargs):
+        if self.cost:
+            azn_rate = float(self.cost)/float(Currency.objects.filter(name="AZN").values_list('rate', flat=True).first())
```
Comment (mmzeynalli) [inline order/models.py]: consider using Decimal for more accuracy

File: order/models.py [hunk 4]
```diff
@@ -8,4 +15,65 @@ class Status(models.Model):
...
+            usd_rate = float(self.cost)/float(Currency.objects.filter(name="USD").values_list('rate', flat=True).first())
+
+        if not self.status:
+            self.status = Status.objects.filter(order = 0).first()
+
+        if self.weight and not self.cost:
+            price =  Tariff.objects.filter(Q(max_weight__gte=self.weight) 
+                & Q(min_weight__lte=self.weight) 
+                & Q(country=self.country)).values_list('base_price', flat=True).first()
```
Comment (mmzeynalli) [inline order/models.py]: Consider having no tariff and having two tariffs for given weight. You should handle them accordingly

File: order/models.py [hunk 5]
```diff
@@ -8,4 +15,65 @@ class Status(models.Model):
...
+            
+            self.cost = price
+            self.cost_azn = azn_rate
+            print(self.cost_azn)
```
Comment (mmzeynalli) [inline order/models.py]: remove prints

File: order/utils.py [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import random
+import string
+
+def generate_tracking_code():
+    return ''.join(random.choices(string.digits, k=13))
+
+def apply_discount(discount, cost):
+    if discount.constant_or_percentage == 'percentage':
```
Comment (mmzeynalli) [inline order/utils.py]: use constants

Comment (mmzeynalli) [reply order/utils.py]: for 'percentage' and 'constant'

File: requirements.txt [hunk 7]
```diff
@@ -12,6 +12,7 @@ click-repl==0.2.0
 coverage==7.2.2
 Django==3.2
 django-ckeditor==6.5.1
+django-field-history==0.8.0
```
Comment (mmzeynalli) [inline requirements.txt]: remove

**Review by mmzeynalli — CHANGES_REQUESTED** [review]

File: order/apps.py (hunk 2 above)
Comment (ShahanAliyev) [reply order/apps.py:9]: the signals do not work without this code

File: order/admin.py [hunk 8]
```diff
@@ -11,3 +11,6 @@ def save_model(self, request, obj, form, change):
         if not obj.user:
             obj.user = request.user
             super().save_model(request, obj, form, change)
+        else:
```
Comment (mmzeynalli) [inline order/admin.py]: remove else, just super().save() outsize of if. You are repeating

**Review by mmzeynalli — CHANGES_REQUESTED** [review]
Also write test units, otherwise I wont approve

**Comment by mmzeynalli** [comment]
It would be better if you fixed your duplicate code that you have in two elifs

**Pull request merged**